src/components/main/HomePage.js

```diff
@@ -26,16 +28,37 @@ const styles = {
 class HomePage extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      waitingTime: 0,
+    };
   }
 
   componentDidMount() {
     const calendarId = this.props.calendar.id;
     const timeMin = moment().startOf('day').format();
     const timeMax = moment().endOf('day').format();
+
     if (calendarId) {
       this.props.actions.loadEvents(calendarId, timeMin, timeMax);
     }
-	}
+
+    const waitingTimer = () => {
+      const { waitingTime } = this.state;
+      if (waitingTime < WAITING_TIME) {
+        this.setState({ waitingTime: waitingTime + 1 });
+      } else {
+        this.props.history.push('/');
+      }
+    };
+
+    this.interval = setInterval(waitingTimer.bind(this), 1000);
+  }
+
+  componentWillUnmount() {
```

Comment:
as I understood, it will refresh the page every 6th second and drop user to '/' route,  as far as the state changes. We can either make a socket connection (maybe it's possible to get smth like notification from API) or make this reload only if the current path is '/'